File: src/System.Windows.Forms.Analyzers.VisualBasic/tests/UnitTests/System.Windows.Forms.Analyzers.VisualBasic.Tests/WFO1001/Data/TypedWithNamespace.vb
Imports System.Windows
Imports System.Diagnostics.CodeAnalysis
Imports System.Reflection.Metadata
Imports System.Runtime.Versioning

Namespace System.Windows.Forms.Analyzers.VisualBasic.Tests

    Friend Class TypedWithNamespace
        Implements Forms.ITypedDataObject

        Public Function GetData(format As String, autoConvert As Boolean) As Object Implements IDataObject.GetData
            Return Nothing
        End Function

        Public Function GetData(format As String) As Object Implements IDataObject.GetData
            Return Nothing
        End Function

        Public Function GetData(format As Type) As Object Implements IDataObject.GetData
            Return Nothing
        End Function

        Public Function GetDataPresent(format As String, autoConvert As Boolean) As Boolean Implements IDataObject.GetDataPresent
            Return False
        End Function

        Public Function GetDataPresent(format As String) As Boolean Implements IDataObject.GetDataPresent
            Return False
        End Function

        Public Function GetDataPresent(format As Type) As Boolean Implements IDataObject.GetDataPresent
            Return False
        End Function

        Public Function GetFormats(autoConvert As Boolean) As String() Implements IDataObject.GetFormats
            Return New String() {"thing1"}
        End Function

        Public Function GetFormats() As String() Implements IDataObject.GetFormats
            Return New String() {"thing1"}
        End Function

        Public Sub SetData(format As String, autoConvert As Boolean, data As Object) Implements IDataObject.SetData
        End Sub

        Public Sub SetData(format As String, data As Object) Implements IDataObject.SetData
        End Sub

        Public Sub SetData(format As Type, data As Object) Implements IDataObject.SetData
        End Sub

        Public Sub SetData(data As Object) Implements IDataObject.SetData
        End Sub

        Public Function TryGetData(Of T)(<DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)> ByRef data As T) As Boolean Implements ITypedDataObject.TryGetData
            Throw New NotImplementedException()
        End Function

        Public Function TryGetData(Of T)(format As String, <DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)> ByRef data As T) As Boolean Implements ITypedDataObject.TryGetData
            Throw New NotImplementedException()
        End Function

        Public Function TryGetData(Of T)(format As String, autoConvert As Boolean, <DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)> ByRef data As T) As Boolean Implements ITypedDataObject.TryGetData
            Throw New NotImplementedException()
        End Function

        Public Function TryGetData(Of T)(format As String, resolver As Func(Of TypeName, Type), autoConvert As Boolean, <DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)> ByRef data As T) As Boolean Implements ITypedDataObject.TryGetData
            Throw New NotImplementedException()
        End Function

    End Class
End Namespace

File: src/System.Windows.Forms.Analyzers.CSharp/tests/UnitTests/WFO1001/Data/TypedWithNamespace.cs
using System.Windows;
using System.Diagnostics.CodeAnalysis;
using System.Reflection.Metadata;
using System.Runtime.Versioning;

namespace System.Windows.Forms.Analyzers.CSharp.Tests;

internal class TypedWithNamespace : Forms.ITypedDataObject
{
    public object? GetData(string format, bool autoConvert) => null;
    public object? GetData(string format) => null;
    public object? GetData(Type format) => null;
    public bool GetDataPresent(string format, bool autoConvert) => false;
    public bool GetDataPresent(string format) => false;
    public bool GetDataPresent(Type format) => false;
    public string[] GetFormats(bool autoConvert) => ["thing1"];
    public string[] GetFormats() => ["thing1"];
    public void SetData(string format, bool autoConvert, object? data) { }
    public void SetData(string format, object? data) { }
    public void SetData(Type format, object? data) { }
    public void SetData(object? data) { }

    public bool TryGetData<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)] T>([MaybeNullWhen(false), NotNullWhen(true)] out T data) =>
    throw new NotImplementedException();
    public bool TryGetData<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)] T>(string format, [MaybeNullWhen(false), NotNullWhen(true)] out T data) =>
        throw new NotImplementedException();
    public bool TryGetData<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)] T>(string format, bool autoConvert, [MaybeNullWhen(false), NotNullWhen(true)] out T data) =>
        throw new NotImplementedException();
    public bool TryGetData<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)] T>(string format, Func<TypeName, Type> resolver, bool autoConvert, [MaybeNullWhen(false), NotNullWhen(true)] out T data) =>
        throw new NotImplementedException();
}
